### PR TITLE
Custom division

### PR DESCRIPTION
**Pull request by rheiland**
Provide functionality for custom cell division (a new `functions.cell_division_function`), plus lineage tracking via `generation` and `parentID` attributes for agents.

### CODE REVIEW

File: modules/PhysiCell_MultiCellDS.cpp [hunk 1]
```diff
@@ -890,6 +890,29 @@ void add_PhysiCell_cells_to_open_xml_pugi_v2( pugi::xml_document& xml_dom, std::
 		data_start_indices.push_back( index ); 
 		cell_data_size += size; 
 		index += size; 
+
+        // generation
+		name = "generation";
```
Comment (drbergman) [inline modules/PhysiCell_MultiCellDS.cpp:895]: when merging with dev-paul, will need (or maybe just want?) to update these with Paul's condensed version of this code there

**Review by drbergman — COMMENTED** [review]
Really excited for this! There's also some changes on where the PhysiCell_constants are defined. Is that related? Or solving some other issue?

**Comment by rheiland** [comment]
> Really excited for this! There's also some changes on where the PhysiCell_constants are defined. Is that related? Or solving some other issue?

This is one reason I've been having mild panic attacks lately :-) over the upcoming merge party. I think I made this branch from my branch which dealt with https://github.com/MathCancer/PhysiCell/pull/248 (I had reasons for doing so at the time). So this touches on merging PRs in a certain order, I think. Or, I could simply update the branch for custom cell division and use the original "constants" code.

**Comment by rheiland** [comment]
Earlier comments about this PR are in the related Issue https://github.com/MathCancer/PhysiCell/issues/269

**Comment by rheiland** [comment]
Closing. The actual custom function for cell division was merged with #285. Perhaps later we can refer to this 273 for an improved cell lineage tracking PR.

**Pull request closed without merging**